controllers/storageclassclaimsecret/storageclassclaimsecret_controller.go

```diff
@@ -0,0 +1,665 @@
+}
+
+var claimPredicate = predicate.Funcs{
+	CreateFunc: func(e event.CreateEvent) bool {
+		return isClaimSecret(e.Object)
+	},
+	DeleteFunc: func(e event.DeleteEvent) bool {
+		return isClaimSecret(e.Object)
+	},
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		return isClaimSecret(e.ObjectNew)
+	},
+	GenericFunc: func(e event.GenericEvent) bool {
+		return isClaimSecret(e.Object)
+	},
+}
+
+func (r *StorageClassClaimSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	enqueueStorageConsumerRequest := handler.EnqueueRequestsFromMapFunc(
+		func(obj client.Object) []reconcile.Request {
+			annotations := obj.GetAnnotations()
+			if annotation, found := annotations[v1alpha1.StorageClassClaimAnnotation]; found {
+				parts := strings.Split(annotation, "/")
+				return []reconcile.Request{{
+					NamespacedName: types.NamespacedName{
+						Namespace: parts[0],
+						Name:      parts[1],
+					},
+				}}
+			}
+			return []reconcile.Request{}
+		})
+	// Compose a predicate that is an OR of the specified predicates
+	//secretsPredicate := util.ComposePredicates(
+	//	predicate.GenerationChangedPredicate{},
+	//	util.MetadataChangedPredicate{},
+	//)
+
+	// As we are not setting the Controller OwnerReference on the ceph
+	// resources we are creating as part of the StorageClassClaim, we need to
+	// set IsController to false to get Reconcile Request of StorageClassClaim
+	// for the owned ceph resources updates.
+	enqueueForNonControllerOwner := &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.StorageClassClaim{}, IsController: false}
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Secret{}, builder.WithPredicates(claimPredicate)).
+		Watches(&source.Kind{Type: &rookCephv1.CephBlockPool{}}, enqueueForNonControllerOwner).
+		Watches(&source.Kind{Type: &rookCephv1.CephFilesystemSubVolumeGroup{}}, enqueueForNonControllerOwner).
+		Watches(&source.Kind{Type: &rookCephv1.CephClient{}}, enqueueForNonControllerOwner).
+		Watches(&source.Kind{Type: &storagev1.StorageClass{}}, enqueueStorageConsumerRequest).
+		Watches(&source.Kind{Type: &snapapi.VolumeSnapshotClass{}}, enqueueStorageConsumerRequest).
+		Complete(r)
+}
+
+func (r *StorageClassClaimSecretReconciler) reconcileProviderPhases() (reconcile.Result, error) {
+	r.log.Info("Running StorageClassClaim Secret controller in Provider Mode")
+
+	r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimInitializing
+
+	gvk, err := apiutil.GVKForObject(&v1alpha1.StorageConsumer{}, r.Client.Scheme())
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get gvk for consumer  %w", err)
+	}
+	// reading storageConsumer Name from storageClassClaimSecret ownerReferences
+	ownerRefs := r.storageClassClaimSecret.GetOwnerReferences()
+	for i := range ownerRefs {
+		if ownerRefs[i].Kind == gvk.Kind {
+			r.storageConsumer = &v1alpha1.StorageConsumer{}
+			r.storageConsumer.Name = ownerRefs[i].Name
+			r.storageConsumer.Namespace = r.OperatorNamespace
+			break
+		}
+	}
```

Comment:
This is on the provider side, so there's no reason to have OwnerRefs on the claim since it's not a real resource. The Secret is the thing that should have OwnerRefs.

Reply:
Ok